src/Microsoft.AspNetCore.Http/Extensions/HttpRequestRewindExtensions.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http.Internal;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Extension methods for enabling buffering in an <see cref="HttpRequest"/>.
+    /// </summary>
+    public static class HttpRequestRewindExtensions
```

Comment:
Why do these methods return the HttpRequest?

Reply:
Works for me either way. Make the call.

I followed the chaining pattern to match `IApplicationBuilder`, `IServiceCollection`, and similar setup extensions. `BufferingHelper` methods also support method chaining.

But, MVC and WebHooks don't require this pattern. Other `HttpRequest` methods and its few extensions aren't following it either.

Reply:
Discussed offline. Will go with `void`.